packages/injectable-react/ogre-tools-injectable-react.d.ts

```diff
@@ -9,28 +9,35 @@ declare module '@ogre-tools/injectable-react' {
 
   export const DiContextProvider: React.Provider<DiContainerProviderProps>;
 
-  export function withInjectables<Dependencies, Props = {}>(
-    Component: React.ElementType<Dependencies & Props>,
-    options: {
-      getProps: (di: DiContainer, props: Props) => Props & Dependencies;
-    },
-  ): React.ForwardRefExoticComponent<
-    Props & React.RefAttributes<any> & { children?: React.ReactNode }
-  >;
-
-  export function withInjectables<Dependencies, Props = {}>(
-    Component: React.ElementType<Dependencies & Props>,
-    options: {
-      getProps: (
-        di: DiContainer,
-        props: Props,
-      ) => PromiseLike<Props & Dependencies>;
-
-      getPlaceholder: () => JSX.Element;
-    },
-  ): React.ForwardRefExoticComponent<
-    Props & React.RefAttributes<any> & { children?: React.ReactNode }
-  >;
+  export interface WithInjectablesOptions<Dependencies extends object, Props extends object> {
+    getProps: (di: DiContainer, props: Props) => Props & Dependencies;
+  }
+
+  export interface WithInjectablesAsyncOptions<Dependencies extends object, Props extends object> {
+    getProps: (di: DiContainer, props: Props) => Promise<Props & Dependencies>;
+    getPlaceholder: () => JSX.Element;
+  }
+
+  export interface WithInjectables {
+    <Dependencies extends object, Props extends object = {}>(
+      Component: React.ElementType<Dependencies & Props>,
+      options: WithInjectablesOptions<Dependencies, Props>,
```

Comment:
Symmetric naming of types required: `WithInjectablesAsyncOptions` should be accompanied with `WithInjectablesSyncOptions`, not `WithInjectablesOptions`.